### ISSUE TITLE
cannot allocate vector of length 1790997466 / loadRcppModules  / loadModule - new windows toolchain

### ISSUE THREAD

**Issue by dtenenba** [RcppCore/Rcpp#462]
Hi,

In R-3.3.0-beta, under the new (gcc/g++ 4.9.3-based) windows [toolchain](https://github.com/rwinlib/r-base), if I try and install from source, under 32-bit R, a package (in this case multicool) that loads Rcpp modules via `loadRcppModules()` or `loadModule()`, I get this:

```
** testing if installed package can be loaded
*** arch - i386
Error in .doLoadActions(where, attach) :
  error in load action .__A__.1 for package multicool: loadModule(module = "Multicool", wh
at = TRUE, env = ns, loadNow = TRUE): Unable to load module "Multicool": cannot allocate v
ector of length 1790997466
Error: loading failed
Execution halted
*** arch - x64
ERROR: loading failed for 'i386'
* removing 'e:/biocbld/bbs-3.3-bioc/R/library/multicool'
```

Interestingly, the CRAN build farm does not seem to have this problem, because the binary version of multicool 0.1-9 loads just fine under i386. However, I and others are having this [issue](https://github.com/sneumann/mzR/issues/36#issuecomment-210107440) on several different machines, with more than one package that uses Rcpp, and the length of the vector it's trying to allocate is always the same (1790997466).

```
> sessionInfo()
R version 3.3.0 beta (2016-04-10 r70462)
Platform: i386-w64-mingw32/i386 (32-bit)
Running under: Windows Server 2008 R2 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=English_United States.1252
[2] LC_CTYPE=English_United States.1252
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C
[5] LC_TIME=English_United States.1252

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base
```

**Comment by dtenenba** [comment]
Forgot to mention my Rcpp version, it was 0.12.4 (the version available on CRAN).
I just tried the latest from github, 0.12.4.4, and it seems this is fixed there. However, the fact that this fix is not available on CRAN is a problem for Bioconductor, because our automated build systems can only use what is available on CRAN. So this is blocking some of our packages from building. Any chance of pushing this fix to CRAN fairly soon?
Thanks.

**Comment by eddelbuettel** [comment]
Please see #460 where this was just reported, and, via nice team effort, resolved.

tl;dr:  mingw wants `long long` for types we need here, CRAN doesn't let us have `long long` and everybody cries as we end up with `int`.  

Fix: make the _client package_ require C++11, and you get (inter alia) `long long` for free.

**Comment by eddelbuettel** [comment]
This is planned to go out just before R 3.3.0 as we need some other fixes.   If you need it sooner, just put Rcpp 0.12.4.4 into a drat or other repo of yours.

**Comment by dtenenba** [comment]
Thanks. I tried to install the bioconductor package mzR (https://bioconductor.org/packages/devel/bioc/src/contrib/mzR_2.5.5.tar.gz) under the latest Rcpp from github and got this:

```
** testing if installed package can be loaded
Found more than one class "Rcpp_Ramp" in cache; using the first, from namespace 'mzR'
Found more than one class "Rcpp_Ramp" in cache; using the first, from namespace 'mzR'
Error : .onLoad failed in loadNamespace() for 'mzR', details:
  call: value[[3L]](cond)
  error: failed to load module Ramp from package mzR
could not find function ".moduleMetaName"
```

Any ideas?

**Comment by eddelbuettel** [comment]
No.  I did a full reverse depends check yesterday and had no packages mailing due to modules load.

For 'Found more than one' the `mzR` package may need to rename a module to not define two with the same name, or otherwise load under the same name.   We had something like that a while back with our unit tests and ensured module / class names to be unique.

**Comment by eddelbuettel** [comment]
Really random guess:  the package is still very old school.  It seems to use the 'shot gun' function `loadRcppModules()`.   

Most (all ?) of mine switched years to `loadModule("NameOfModuleClass", TRUE)`.  You may want to try that.

**Comment by dtenenba** [comment]
That works! Thanks.

**Comment by eddelbuettel** [comment]
Yes, sure.  

Give the author my regards; I cannot recall how many time I recommended they overhaul the package :cry:

**Comment by thirdwing** [comment]
Actually `mzR` is developed by me around two years ago...

I don't have much time to update it since then.
